tfregistry/provider_test.go

```diff
@@ -0,0 +1,94 @@
+package tfregistry
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestProviderLatest(t *testing.T) {
+	cases := []struct {
+		desc string
+		req  *ProviderLatestRequest
+		ok   bool
+		code int
+		res  string
+		want *ProviderLatestResponse
+	}{
+		{
+			desc: "simple",
+			req: &ProviderLatestRequest{
+				Namespace: "hashicorp",
+				Type:      "aws",
+			},
+			ok:   true,
+			code: 200,
+			res:  `{"version": "3.7.0", "versions": ["3.7.0", "3.6.0", "3.5.0"]}`,
```

Comment:
The actual API response is ASC order. So I think `["3.5.0", "3.6.0", "3.7.0"]` is better than DESC for test data.